### PR TITLE
extensive refactoring

### PR DESCRIPTION
**Pull request by smoebody**
* adds function docs
* utilises built in functions for sanitising
* caches login responses as promise
* caches getClientList response as promise
* introduces getClientByMac(), getIpTraffic()
* introduces custom exceptions
* changes code style, updates packages
* adds multiple node build targets, removes v16

### CODE REVIEW

**Comment by smoebody** [comment]
@theimo1221 Please can you have a look and tell me whether you are ok with this?

**Comment by theimo1221** [comment]
Thanks @smoebody 

Personally I'd prefer the usage of semicolons like microsoft themself (e.g.`https://github.com/microsoft/TypeScript/tree/main`) as this reduces pitfalls for non experienced coders.

File: src/Models/iptraffic.ts
Comment (theimo1221) [inline src/Models/iptraffic.ts:1]: Why is this not a normal interface and or class?

**Review by theimo1221 — CHANGES_REQUESTED** [review]

File: src/Models/iptraffic.ts
Comment (smoebody) [reply src/Models/iptraffic.ts:1]: I don't understand, why should it be? Types are totally sufficient and there is no need for interfaces or even a class. 
Maybe it should not be in Models. But then there is also the enum `DeviceAction` and other interface-only declarations...

Comment (theimo1221) [reply src/Models/iptraffic.ts:1]: I might be a more conservative typescript developer, thus I discourage usage of types and especially nested types like `clients: (Traffic & { mac: string })[]`.
Additionally in my opinion there should only be one type/class/interface per file

Comment (smoebody) [reply src/Models/iptraffic.ts:1]: I changed it to be an interface

**Review by theimo1221 — APPROVED** [review]

**Pull request merged**